fish_prompt.fish

```diff
@@ -228,6 +237,38 @@ end
 # Theme components
 # ===========================
 
+function __bobthefish_prompt_vagrant -d 'Display Vagrant status'
+  [ "$theme_display_vagrant" != 'yes' ]; and return
+  which VBoxManage >/dev/null 2>&1; and set -l __vbox_installed yes
+  if [ -f Vagrantfile -a "$__vbox_installed" = 'yes' ]
+    # Get machine UUIDs
+    set -e __vagrant_ids
+    set -e __vagrant_statuses
+    for m in .vagrant/machines/**/id
+      set -l __machine_id (cat $m)
+      set __vagrant_ids $__vagrant_ids $__machine_id
+    end
+    for i in $__vagrant_ids
+      set -l __vm_status (VBoxManage showvminfo --machinereadable $i 2>/dev/null | grep 'VMState=' | tr -d '"' | cut -d '=' -f 2)
+      set __vagrant_statuses "$__vagrant_statuses<$__vm_status>"
+    end
+    # Transform statuses to gliphs
+    set __vagrant_statuses ( echo -n $__vagrant_statuses | \
+      sed -e "s#<running>#$__bobthefish_vagrant_running#g" | \
```

Comment:
No need for the pipes here; you can perform multiple sed replacements in one go.

```
$ echo '<wat>' | sed -e 's#w#th#g' -e 's#that#this#g' -e 's#<this>#[wheee]#g'
[wheee]
```